secrets/README.md

```diff
@@ -0,0 +1,18 @@
+# Secret Management
```

Comment:
Nice!

One thing to discuss/put in README.

How it relates to https://github.com/prometheus/common/blob/d80d8544703e59a080a204b6f7429ac6561fb24f/config/config.go#L28

What would be the plan forward? Deprecate `config.Secret`?